lib/excon/ssl_socket.rb

```diff
@@ -66,16 +66,22 @@ def initialize(data = {})
       end
 
       # maintain existing API
-      certificate_data = @data[:client_cert] || @data[:certificate_path]
-      private_key_data = @data[:client_key] || @data[:private_key_path]
-      private_key_pass = @data[:client_key_pass] || @data[:private_key_pass]
+      client_cert_data = if cert_data = @data[:client_cert_data]
+                           cert_data
+                         elsif path = (@data[:client_cert] || @data[:certificate_path])
+                           File.read path
+                         end
+
+      private_key_data = if key_data = @data[:private_key_data]
```

Comment:
I think `private_key` is the legacy key here, actually. So this should probably actually be `client_key_data` for consistency. Does that make sense?